src/psyclone/domain/nemo/transformations/nemo_outerarrayrange2loop_trans.py

```diff
@@ -161,29 +157,24 @@ def validate(self, node, options=None):
         # Am I an assignment node?
         if not isinstance(node, Assignment):
             raise TransformationError(
-                "Error in NemoOuterArrayRange2LoopTrans transformation. The "
-                "supplied node argument should be a PSyIR Assignment, but "
-                "found '{0}'.".format(type(node).__name__))
+                f"Error in NemoOuterArrayRange2LoopTrans transformation. The "
+                f"supplied node argument should be a PSyIR Assignment, but "
+                f"found '{type(node).__name__}'.")
 
         # Is the LHS an array reference?
-        if not isinstance(node.lhs, (ArrayReference,
-                                     ArrayOfStructuresReference)):
+        if not (isinstance(node.lhs, Reference) and node.lhs.walk(ArrayMixin)):
             raise TransformationError(
-                "Error in NemoOuterArrayRange2LoopTrans transformation. The "
-                "supplied assignment node should have either an ArrayReference"
-                " or an ArrayOfStructuresReference node on its lhs but found "
-                "'{0}'.".format(type(node.lhs).__name__))
-        array_reference = node.lhs
+                f"Error in NemoOuterArrayRange2LoopTrans transformation. The "
+                f"supplied assignment node should be a Reference that contains"
+                f" an ArrayMixin somewhere in the expression, but found "
+                f"'{node.lhs}'.")
         # Has the array reference got a range?
-        try:
-            _ = get_outer_index(array_reference)
-        except IndexError as error:
-            message = (
-                "Error in NemoOuterArrayRange2LoopTrans transformation. The "
-                "LHS of the supplied assignment node should be an "
-                "ArrayReference/ArrayOfStructuresReference node containing at "
-                "least one Range node but there are none.")
-            six.raise_from(TransformationError(message), error)
+        if not node.lhs.walk(Range):
+            raise TransformationError(
+                f"Error in NemoOuterArrayRange2LoopTrans transformation. The "
+                f"LHS of the supplied assignment node should be an expression "
+                f"with an array that has a Range node, but found '{node.lhs}'."
+                )
```

Comment:
Please could you put this closing parenthesis on the previous line. Is the output of `{node.lhs}` suitably user friendly? (I expect I'll find out when I get to the tests.)

Reply:
Done just moving the parenthesis for now.